hooks/hookexecution/executor.go

```diff
@@ -0,0 +1,134 @@
+package hookexecution
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/hooks"
+	"github.com/prebid/prebid-server/hooks/hookstage"
+)
+
+const (
+	EndpointAuction = "/openrtb2/auction"
+	EndpointAmp     = "/openrtb2/amp"
+)
+
+type StageExecutor interface {
+	ExecuteEntrypointStage(req *http.Request, body []byte) ([]byte, *RejectError)
+	ExecuteRawAuctionStage(body []byte) ([]byte, *RejectError)
+	ExecuteProcessedAuctionStage(req *openrtb2.BidRequest) *RejectError
+	ExecuteBidderRequestStage(req *openrtb2.BidRequest, bidder string) *RejectError
+	ExecuteRawBidderResponseStage(response *adapters.BidderResponse) *RejectError
+	ExecuteAllProcessedBidResponsesStage(responses []*adapters.BidderResponse) //TODO: check that responses is the necessary param
+	ExecuteAuctionResponseStage(response *openrtb2.BidResponse)
+}
+
+type HookStageExecutor interface {
+	StageExecutor
+	SetAccount(account *config.Account)
+	GetOutcomes() []StageOutcome
+}
```

Comment:
Do we need two separate interfaces? If we don't see any use for `StageExecutor` other than embedding it in `HookStageExecutor` then it might be less confusing if these two interfaces are combined.

Reply:
We just decided to segregate them and use different interfaces in the endpoint deps (HookStageExecutor) and internal functions (StageExecutor) since we did not like to have SetAccount and GetOutcomes functions together with Execute{Stage} in one interface but it is not a problem to use one interface if you think it is confusing

Reply:
Waiting for your decision whether to change it or leave it

Reply:
Let's leave this as is for now and revisit during the formal review in the PBS repo.